TestResultSummaryService/plugins/DataManagerPerf.js

```diff
@@ -0,0 +1,65 @@
+const ObjectID = require( 'mongodb' ).ObjectID;
+const BenchmarkMath = require( '../perf/BenchmarkMathCalculation' );
+const math = require('mathjs');
+
+module.exports.onBuildDone = async (task, { testResultsDB, logger }) => {
+    logger.debug("onBuildDone", task.buildName);
+    if ( task.hasChildren ) {
+        const parentId = task._id;
+        const childBuildList = await testResultsDB.getData({parentId}).toArray();
+        if ( childBuildList && childBuildList.length > 0 ) {
+            const parentAggregationStructure = [];
+            let benchmarkName, benchmarkVariant, benchmarkProduct;
+            const childrenAggData = [];
+            const metricsCollection = {};
+            //loop into the child build list
+            for ( let j = 0; j < childBuildList.length; j++ ) {
```

Comment:
Use the following, then we can remove the check in line 10:
`for ( let childBuild of childBuildList ) {`

Reply:
updated!